vyper/ast/nodes.py

```diff
@@ -1057,20 +1057,32 @@ def _op(self, left, right):
         if not right:
             raise ZeroDivisionException("Division by zero")
 
-        if isinstance(left, decimal.Decimal):
-            value = left / right
-            if value < 0:
-                # the EVM always truncates toward zero
-                value = -(-left / right)
-            # ensure that the result is truncated to MAX_DECIMAL_PLACES
-            return value.quantize(
-                decimal.Decimal(f"{1:0.{MAX_DECIMAL_PLACES}f}"), decimal.ROUND_DOWN
-            )
-        else:
-            value = left // right
-            if value < 0:
-                return -(-left // right)
-            return value
+        if not isinstance(left, decimal.Decimal):
+            raise UnfoldableNode("Cannot use `/` on non-decimals (did you mean `//`?)")
+
+        value = left / right
+        if value < 0:
+            # the EVM always truncates toward zero
+            value = -(-left / right)
+        # ensure that the result is truncated to MAX_DECIMAL_PLACES
+        return value.quantize(decimal.Decimal(f"{1:0.{MAX_DECIMAL_PLACES}f}"), decimal.ROUND_DOWN)
+
+
+class FloorDiv(VyperNode):
+    __slots__ = ()
+    _description = "integer division"
```

Comment:
We can consider changing the description for `Div` node to `decimal division` for better error messages.

```
@external
def foo() -> decimal:
    x: int128 = as_wei_value(5, "finney")
    y: int128 = block.timestamp + 50
    return x // y
```
```
vyper.exceptions.InvalidOperation: Cannot perform division on uint256
E       
E         function "foo", line 6:11 
E              5     y: uint256 = block.timestamp + 50 - block.timestamp
E         ---> 6     return x / y
E         ------------------^
E              7
```

Reply:
yep i agree